src/adcm_pytest_plugin/fixtures.py

```diff
@@ -311,3 +314,36 @@ def sdk_client_ss(adcm_ss: ADCM, adcm_api_credentials) -> ADCMClient:
 def cmd_opts(request):
     """Returns pytest request options object"""
     return request.config.option
+
+
+@allure.title("Create provider")
+@pytest.fixture(scope="session")
+def provider_ss(sdk_client_ss) -> Provider:
+    bundle = sdk_client_ss.upload_from_fs(utils.get_data_dir(__file__, "provider"))
+    return bundle.provider_prototype().provider_create("Some provider")
+
+
+@allure.title("Create a cluster with service")
+@pytest.fixture(scope="session")
+def cluster_with_service_ss(sdk_client_ss) -> Cluster:
+    bundle = sdk_client_ss.upload_from_fs(utils.get_data_dir(__file__, "cluster_with_service"))
+    cluster = bundle.cluster_prototype().cluster_create(name="Cluster with services")
+    return cluster
+
+
+@allure.title("ADCM Client with ADCM objects")
+@pytest.fixture(scope="session")
+def sdk_client_with_objects_ss(cluster_with_service_ss: Cluster, provider_ss: Provider):
+    """Returns ADCMClient with ADCM objects"""
+    service_first = cluster_with_service_ss.service_add(name="Dummy service")
+    service_second = cluster_with_service_ss.service_add(name="Second service")
+    host_first = provider_ss.host_create("host_in_cluster")
+    host_second = provider_ss.host_create("host_in_cluster_second")
+    cluster_with_service_ss.host_add(host_first)
+    cluster_with_service_ss.host_add(host_second)
+    cluster_with_service_ss.hostcomponent_set((host_first, service_first.component(name="first")),
+                                              (host_second, service_first.component(name="second")),
+                                              (host_second, service_second.component(name="third")))
+    task = cluster_with_service_ss.action(name="action_on_cluster").run()
+    task.wait()
+    return sdk_client_ss
```

Comment:
Reformat these fixtures to be a single private function (or a bunch of functional but with a single entry point function)

Reply:
done